tests/e2e/test_rosacli_cluster.go

```diff
@@ -569,3 +611,74 @@ var _ = Describe("Classic cluster creation negavite testing",
 			Expect(out.String()).To(ContainSubstring("to create compatible roles and try again"))
 		})
 	})
+
+var _ = Describe("HCP cluster creation negavite testing",
+	labels.Day1Negative,
+	labels.FeatureCluster,
+	func() {
+		defer GinkgoRecover()
+
+		var (
+			clusterID      string
+			rosaClient     *rosacli.Client
+			clusterService rosacli.ClusterService
+		)
+		BeforeEach(func() {
+			By("Get the cluster")
+			clusterID = config.GetClusterID()
+			Expect(clusterID).ToNot(Equal(""), "ClusterID is required. Please export CLUSTER_ID")
+
+			By("Init the client")
+			rosaClient = rosacli.NewClient()
+			clusterService = rosaClient.Cluster
+		})
+
+		It("create HCP cluster with network type validation can work well via rosa cli - [id:73725]", labels.Medium, func() {
+			isHostedCP, err := clusterService.IsHostedCPCluster(clusterID)
+			Expect(err).To(BeNil())
+
+			rosalCommand, err := config.RetrieveClusterCreationCommand(ciConfig.Test.CreateCommandFile)
+			Expect(err).To(BeNil())
+
+			if !isHostedCP {
+				By("Create non-HCP cluster with --no-cni flag")
+				clusterName := common.GenerateRandomName("classic-73725", 2)
+				operatorPrefix := common.GenerateRandomName("classic-oper", 2)
+				replacingFlags := map[string]string{
+					"-c":                     clusterName,
+					"--cluster-name":         clusterName,
+					"--domain-prefix":        clusterName,
+					"--operator-role-prefix": operatorPrefix,
+				}
+				rosalCommand.ReplaceFlagValue(replacingFlags)
+				rosalCommand.AddFlags("--dry-run", "--no-cni", "-y")
+				output, err := rosaClient.Runner.RunCMD(strings.Split(rosalCommand.GetFullCommand(), " "))
+				Expect(err).To(HaveOccurred())
+				Expect(output.String()).To(ContainSubstring("ERR: Disabling CNI is supported only for Hosted Control Planes"))
+			} else {
+				By("Create HCP cluster with --no-cni and \"--network-type={OVNKubernetes, OpenshiftSDN}\" at the same time")
+				clusterName := common.GenerateRandomName("cluster-71946", 2)
+				operatorPrefix := common.GenerateRandomName("cluster-oper", 2)
+
+				replacingFlags := map[string]string{
+					"-c":                     clusterName,
+					"--cluster-name":         clusterName,
+					"--domain-prefix":        clusterName,
+					"--operator-role-prefix": operatorPrefix,
+				}
+				rosalCommand.ReplaceFlagValue(replacingFlags)
+				rosalCommand.AddFlags("--dry-run", "--no-cni", "--network-type='{OVNKubernetes,OpenshiftSDN}'", "-y")
+				output, err := rosaClient.Runner.RunCMD(strings.Split(rosalCommand.GetFullCommand(), " "))
+				Expect(err).To(HaveOccurred())
+				Expect(output.String()).To(ContainSubstring("ERR: Expected a valid network type. Valid values: [OpenShiftSDN OVNKubernetes]"))
```

Comment:
I think  we can add one step with valid --network-type=OVNKubernetes and --no-cni

Reply:
But will it not be positive?

Reply:
done